common/tests/FileUtilsTest.java

```diff
@@ -95,18 +97,698 @@ public void testFolderExistsIsTrueWhenFolder() throws IOException {
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testReadPrototxtWithEnumType() throws IOException {
+    if (fileSystemName.equals("Windows")) {
+      return;
+    }
+    Path testPath = fileSystem.getPath(TEST_PROTOTXT_FILE_PATH);
+    Files.write(testPath, ImmutableList.of(
+            "favorite: GREEN"),
+            UTF_8);
+    EnumMessage actual =
+            (EnumMessage) fileUtils.readPrototxt(TEST_PROTOTXT_FILE_PATH, EnumMessage.newBuilder());
+    EnumMessage expected = EnumMessage.newBuilder()
+            .setFavorite(EnumMessage.Color.GREEN)
+            .build();
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testReadPrototxtWithRepeatedEnumType() throws IOException {
+    if (fileSystemName.equals("Windows")) {
+      return;
+    }
+    Path testPath = fileSystem.getPath(TEST_PROTOTXT_FILE_PATH);
+    Files.write(testPath, ImmutableList.of(
+            "additional: BLUE",
+            "additional: RED"),
+            UTF_8);
+    EnumMessage actual =
+            (EnumMessage) fileUtils.readPrototxt(TEST_PROTOTXT_FILE_PATH, EnumMessage.newBuilder());
+    EnumMessage expected = EnumMessage.newBuilder()
+            .addAllAdditional(ImmutableList.of(EnumMessage.Color.BLUE, EnumMessage.Color.RED))
+            .build();
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testReadPrototxtWithNestedType() throws IOException {
+    if (fileSystemName.equals("Windows")) {
+      return;
+    }
+    Path testPath = fileSystem.getPath(TEST_PROTOTXT_FILE_PATH);
+    Files.write(testPath, ImmutableList.of(
+            "child {",
+            "name: \"foo\"",
+            "}"),
+            UTF_8);
+    Parent actual =
+            (Parent) fileUtils.readPrototxt(TEST_PROTOTXT_FILE_PATH, Parent.newBuilder());
+    Parent expected = Parent.newBuilder()
+            .setChild(Parent.Child.newBuilder().setName("foo").build())
+            .build();
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testReadPrototxtWithMapType() throws IOException {
+    if (fileSystemName.equals("Windows")) {
+      return;
+    }
+    Path testPath = fileSystem.getPath(TEST_PROTOTXT_FILE_PATH);
+    Files.write(testPath, ImmutableList.of(
+            "map_field {",
+            "key: \"foo\"",
+            "value: 123",
+            "}"),
+            UTF_8);
+    MapMessage actual =
+            (MapMessage) fileUtils.readPrototxt(TEST_PROTOTXT_FILE_PATH, MapMessage.newBuilder());
+    MapMessage expected = MapMessage.newBuilder()
+            .putMapField("foo", 123)
+            .build();
+    assertEquals(expected, actual);
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testReadPrototxtUncheckedWithException() {
+    if (fileSystemName.equals("Windows")) {
+      throw new RuntimeException();
+    }
+    fileUtils.readPrototxtUnchecked("", EnumMessage.newBuilder());
+  }
+
+  @Test
+  public void testWritePrototxtWithScalarTypes() throws IOException {
```

Comment:
General note about the prototxt and protoBinary tests - we're testing FileUtils, and most of these tests are actually testing the proto library. What we care about now, is that a file was read and written properly.
Let's suppose we have a single test here that tests that *some* prototxt was correctly read. Now let's assume there's some bug in reading a prototxt. It's very likely this bug will be in the proto library and not in FileUtils.

So while all these tests are indeed testing important things (i.e if there's a bug in reading a proto enum, they will fail), they should be in the proto library (which already has these tests probably, and in FileUtils tests, we assume the proto library is perfect).

So I'd like you to go over all of them, and see which ones you think should stay. I think that maybe 1 to read, 1 to write, 1 for prototxt and 1 for protoBinary (so overall 2X2=4), but I might be missing something that needs to be tested. So I want you to think about it. Which of these tests are important to FileUtils (and which are important, but should be in proto library).